### PR TITLE
Add publicCommandName for command interface, add PublicCommandName enum

### PR DESCRIPTION
**Pull request by Yuqi-Du**
Separated from #1397, this PR is adding a new method for command interface, which returns the public command name for the implementation command class.
e.g.
FindKeyspacesCommand publicCommandName -> findKeyspaces 
CreateCollectionCommand publicCommandName -> createCollection

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Command.java [hunk 1]
```diff
@@ -27,4 +27,51 @@
...
+
+  /**
+   * commandName that refers to the public command name
+   *
+   * <p>e.g. FindKeyspacesCommand publicCommandName -> findKeyspaces. CreateCollectionCommand
+   * publicCommandName -> createCollection
+   */
+  PublicCommandName publicCommandName();
+
+  /** Enum class for API public command name This is what user uses for command json body. */
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Command.java:40]: nit: missing "." -> "... command name. This is... "

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Command.java [hunk 2]
```diff
@@ -27,4 +27,51 @@
...
+  enum PublicCommandName {
+    addIndex("addIndex"),
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Command.java:42]: Java naming convention typically has these as upper-snake case, so `ADD_INDEX("addIndex")`?

**Review by tatu-at-datastax — COMMENTED** [review]
I think enum names should follow standard Java naming convention: to discuss.

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Command.java (hunk 1 above)
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Command.java:40]: thanks

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Command.java (hunk 2 above)
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Command.java:42]: Thanks!

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Command.java [hunk 3]
```diff
@@ -27,4 +27,51 @@
...
-public interface Command {}
+public interface Command {
+
+  /**
+   * commandName that refers to the public command name
+   *
+   * <p>e.g. FindKeyspacesCommand publicCommandName -> findKeyspaces. CreateCollectionCommand
+   * publicCommandName -> createCollection
+   */
+  PublicCommandName publicCommandName();
```
Comment (amorton) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Command.java:38]: nit: we can just call the enum CommandName and have Command.commandName()

on the enum can we change `publicCommandName` and it's accessor to `apiName` we have that in other places.

**Review by tatu-at-datastax — APPROVED** [review]
LGTM

**Review by amorton — CHANGES_REQUESTED** [review]
just a rename, and +1 on using correct JAVA_ENUM standard for names.

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Command.java (hunk 3 above)
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Command.java]: Thanks!

**Review by amorton — APPROVED** [review]
looks good to me

**Pull request merged**